internal/csi-addons/networkfence/fencing_test.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2022 The Ceph-CSI Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package networkfence
+
+import (
+	"sort"
+	"testing"
+)
+
+func TestGetIPRange(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		cidr        string
+		expectedIPs []string
+	}{
+		{
+			cidr:        "192.168.1.0/30",
+			expectedIPs: []string{"192.168.1.0", "192.168.1.1"},
+		},
+		{
+			cidr:        "10.0.0.0/30",
+			expectedIPs: []string{"10.0.0.0", "10.0.0.1", "10.0.0.2", "10.0.0.3"},
```

Comment:
These validations are incomplete. A /30 network should always return the same amount of IP-addresses. In the 1st case it returns only 2, in the second case it returns 4. There should be a check that **all** IP-addresses are returned, none may be missing, and not one more than expected.

Reply:
Oh no, that was an overlook :(  I forgot to update the CIDR to `/31` and just updated the expected range. Corrected now. Thanks for catching that :)

Reply:
Also added the suggested check